pkg/recipes/terraform/module.go

```diff
@@ -53,13 +50,21 @@ type moduleInspectResult struct {
 // localModuleName is the name of the module specified in the configuration used to download the module.
 // It uses terraform-config-inspect to load the module from the directory. An error is returned if the module
 // could not be loaded.
-func inspectModule(workingDir, localModuleName string) (*moduleInspectResult, error) {
+func inspectModule(workingDir, localModuleName string, templatePath string) (*moduleInspectResult, error) {
+	moduleRootDir := filepath.Join(".terraform/modules", localModuleName)
 	result := &moduleInspectResult{ContextVarExists: false, RequiredProviders: []string{}, ResultOutputExists: false, Parameters: map[string]any{}}
 
 	// Modules are downloaded in a subdirectory in the working directory.
 	// Name of the module specified in the configuration is used as subdirectory name.
 	// https://developer.hashicorp.com/terraform/tutorials/modules/module-use#understand-how-modules-work
-	mod, diags := tfconfig.LoadModule(filepath.Join(workingDir, moduleRootDir, localModuleName))
+	//
+	// If the template path is for a submodule, we'll add the submodule path to the module root directory.
+	subModDex := strings.LastIndex(templatePath, "//")
```

Comment:
I believe that line 62-66 could be extracted to a function for more readability. It could also be easier to just unit test that function for different use cases. Something like **getModulePath**? You know the context much better...

Reply:
We can also add a good documentation to that function.

Reply:
Ended up using an existing TF function so these lines won't be needed